ironfish/src/fileStores/keyStore.ts

```diff
@@ -48,10 +48,15 @@ export class KeyStore<TSchema extends Record<string, unknown>> {
 
     // Validate file store if we have a schema
     if (this.schema) {
-      const { error } = await YupUtils.tryValidate(this.schema, data)
+      const { error, result } = await YupUtils.tryValidate(this.schema, data)
+
       if (error) {
         throw new Error(error.message)
       }
+
+      if (result) {
+        Object.assign(data, result)
```

Comment:
Also I think we might need to check if `data === null` before this. If the file doesn't exist.

Reply:
added check